lib/src/default_revset_engine.rs

```diff
@@ -384,16 +384,16 @@ pub fn evaluate<'index>(
     repo: &'index dyn Repo,
     expression: &RevsetExpression,
 ) -> Result<Box<dyn Revset<'index> + 'index>, RevsetError> {
-    let revset_impl = evaluate_impl(repo, expression)?;
-    Ok(Box::new(revset_impl))
+    let internal_revset = internal_evaluate(repo, expression)?;
+    Ok(Box::new(RevsetImpl::new(internal_revset)))
```

Comment:
Now we can remove `impl<'index> ToPredicateFn<'index> for RevsetImpl<'index>` (line ~70)

Reply:
Good point. Done.